packages/project-access/src/project/cap.ts

```diff
@@ -0,0 +1,118 @@
+import { join } from 'path';
+import type { csn } from '@sap/cds/apis/csn';
+import { FileName } from '@sap-ux/project-types';
+import type { Package } from '@sap-ux/project-types';
+import { fileExists, readJSON } from '../file';
+import { loadModuleFromProject } from './moduleLoader';
+
+/**
+ * Returns true if the project is either a CAP Node.js or a CAP Java project.
+ *
+ * @param projectRoot - the root path of the project
+ * @param [packageJson] - optional: the parsed package.json object
+ * @returns - true if the project is a CAP project; false otherwise
+ */
+export async function isCapProject(projectRoot: string, packageJson?: Package): Promise<boolean> {
+    return (await isCapNodeJsProject(projectRoot, packageJson)) || (await isCapJavaProject(projectRoot));
+}
+
+/**
+ * Returns true if the project is a CAP Node.js project.
+ *
+ * @param projectRoot - the root path of the project
+ * @param [packageJson] - the parsed package.json object
+ * @returns - true if the project is a CAP Node.js project
+ */
+export async function isCapNodeJsProject(projectRoot: string, packageJson?: Package): Promise<boolean> {
+    // Parse package.json file if not provided
+    if (!packageJson) {
+        try {
+            packageJson = await readJSON<Package>(join(projectRoot, FileName.Package));
+        } catch {
+            // Ignore errors while reading the package.json file
+        }
+    }
+    return !!(packageJson?.cds || packageJson?.dependencies?.['@sap/cds']);
+}
+
+/**
+ * Returns true if the project is a CAP Java project.
+ *
+ * @param projectRoot - the root path of the project
+ * @returns - true if the project is a CAP project
+ */
+export async function isCapJavaProject(projectRoot: string): Promise<boolean> {
+    return fileExists(join(projectRoot, 'srv', 'src', 'main', 'resources', 'application.yaml'));
+}
+
+export interface CapCustomPaths {
+    app: string;
+    db: string;
+    srv: string;
+}
+
+interface CdsEnvFolders {
```

Comment:
Shouldn't that also move to the `project-types` project?

Reply:
While `CdsEnvFolders` is used local in this file only, you are right for `CapCustomPaths`. I move it to `project-types`.